--- a/lib/plugins/sfPropelPlugin/lib/vendor/propel/util/BasePeer.php	2010-03-16 00:09:22.000000000 +0100
+++ b/lib/plugins/sfPropelPlugin/lib/vendor/propel/util/BasePeer.php	2018-09-27 10:01:01.966687223 +0200
@@ -155,7 +155,13 @@
 
 			// Execute the statement.
 			try {
-				$sql = "DELETE FROM " . $tableName . " WHERE " .  implode(" AND ", $whereClause);
+				$sql = "DELETE FROM ";
+				if ($db->useQuoteIdentifier()) {
+ 					$sql .= $db->quoteIdentifierTable($tableName);
+				} else {
+					$sql .= $tableName;
+				}
+				$sql .= " WHERE " .  implode(" AND ", $whereClause);
 				$stmt = $con->prepare($sql);
 				self::populateStmtValues($stmt, $selectParams, $dbMap, $db);
 				$stmt->execute();
@@ -192,7 +198,7 @@
 	public static function doDeleteAll($tableName, PropelPDO $con)
 	{
 		try {
-			$sql = "DELETE FROM " . $tableName;
+			$sql = "DELETE FROM `" . $tableName."`";
 			$stmt = $con->prepare($sql);
 			$stmt->execute();
 			return $stmt->rowCount();
@@ -279,9 +285,13 @@
 			if ($adapter->useQuoteIdentifier()) {
 				$columns = array_map(array($adapter, 'quoteIdentifier'), $columns);
 			}
-
-			$sql = 'INSERT INTO ' . $tableName
-			. ' (' . implode(',', $columns) . ')'
+			$sql = 'INSERT INTO ';
+			if ($adapter->useQuoteIdentifier()) {
+				$sql .= $adapter->quoteIdentifier($tableName);
+			} else {
+				$sql .= $tableName;
+			}
+			$sql .= ' (' . implode(',', $columns) . ')'
 			. ' VALUES (';
 			// . substr(str_repeat("?,", count($columns)), 0, -1) . 
 			for($p=1, $cnt=count($columns); $p <= $cnt; $p++) {
@@ -353,8 +353,13 @@
 
 			$stmt = null;
 			try {
-
-				$sql = "UPDATE " . $tableName . " SET ";
+				$sql = "UPDATE ";
+                                if ($db->useQuoteIdentifier()) {
+                                        $sql .= $db->quoteIdentifierTable($tableName);
+                                } else {
+                                        $sql .= $tableName;
+                                }
+                                $sql .= " SET ";
 				$p = 1;
 				foreach ($updateTablesColumns[$tableName] as $col) {
 					$updateColumnName = substr($col, strrpos($col, '.') + 1);
@@ -886,7 +891,11 @@
 					$condition .= ' AND ';
 				}
 			}
-
+			if ($db->useQuoteIdentifier()) {
+				$_rightTable = $db->quoteIdentifierTable($rightTable);
+			} else {
+				$_rightTable = $rightTable;
+			}
 			// add 'em to the queues..
 			if ($joinType = $join->getJoinType()) {
 			  // real join
@@ -894,7 +903,7 @@
 					$fromClause[] = $leftTable . $leftTableAlias;
 				}
 				$joinTables[] = $rightTable . $rightTableAlias;
-				$joinClause[] = $join->getJoinType() . ' ' . $rightTable . $rightTableAlias . " ON ($condition)";
+				$joinClause[] = $join->getJoinType() . ' ' . $_rightTable . $rightTableAlias . " ON ($condition)";
 			} else {
 			  // implicit join, translates to a where
 				$fromClause[] = $leftTable . $leftTableAlias;
